pkg/diagnostics/workflow_metrics.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diagnostics
+
+import (
+	"context"
+
+	diagUtils "github.com/dapr/dapr/pkg/diagnostics/utils"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+var (
+	elapsedKey = tag.MustNewKey("elapsed")
+
+	reminderTypeKey = tag.MustNewKey("reminder_type")
+
+	executionTypeKey = tag.MustNewKey("execution_type")
+)
+
+const (
+	StatusSuccess   = "success"
+	StatusFailed    = "failed"
+	StatusRetryable = "retryable"
+	CreateWorkflow  = "create_workflow"
+	GetWorkflow     = "get_workflow"
+	AddEvent        = "add_event"
+	PurgeWorkflow   = "purge_workflow"
+	Activity        = "activity"
+	Workflow        = "workflow"
+	WorkflowEvent   = "event"
+	Timer           = "timer"
+)
+
+type workflowMetrics struct {
+	// Total Successful and Failed workflow operation requests
+	workflowOperationsCount *stats.Int64Measure
+	// Workflow operation request's Response latency
+	workflowOperationsLatency *stats.Float64Measure
+	// Total workflow/activity reminders created
+	workflowRemindersCount *stats.Int64Measure
+	// Total successful/failed workflow/activity executions
+	workflowExecutionCount *stats.Int64Measure
+	// Time taken to run a workflow to completion
+	workflowExecutionLatency *stats.Float64Measure
+	// Latency between execution request and actual execution
+	workflowSchedulingLatency *stats.Float64Measure
```

Comment:
This we are not implementing in the current PR right?

Reply:
We can probably add this. When a workflow creation request comes up, a historyEvent is created and sent: https://github.com/dapr/dapr/blob/ed3417223c5ea4654f38df1fc5abee93e52cea6e/pkg/runtime/wfengine/workflow.go#L173 which contains the timeStamp. And when it is actually scheduled i.e. reminders gets invoked, the same historyEvent is there in workflow inbox: https://github.com/dapr/dapr/blob/ed3417223c5ea4654f38df1fc5abee93e52cea6e/pkg/runtime/wfengine/workflow.go#L332. We can use this to calculate the Scheduling latency. 

I saw it marked as requiring runtime changes in the issue: https://github.com/dapr/dapr/issues/7109. @shubham1172 Which runtime changes you meant?

Reply:
It would actually be the difference in creating a reminder and invoking that reminder. Idk if there is a better way to do that...

Reply:
Hi @shivamkm07, I meant the runtime changes mentioned in the issue.
![image](https://github.com/dapr/dapr/assets/22132944/e3afa11a-74df-4ec4-b094-a8d31ae450c3)

The idea was to use Actors reminder payload to calculate when the reminder was fired vs when it was executed.
We can alternatively explore the other way too, by looking into the `historyEvent`. Does it also work for activities along with workflow?

Reply:
That makes sense. HistoryEvent approach might not be the correct one as I pointed out above. We can explore this later ig. It should be out of scope of this PR then.

Reply:
sure. I will remove it for now.

Reply:
removed `workflowSchedulingLatency` metric.